wasm_samples/multisig_register/src/lib.rs

```diff
@@ -22,20 +25,19 @@ getrandom::register_custom_getrandom!(iroha_trigger::stub_getrandom);
 const WASM: &[u8] = core::include_bytes!(concat!(core::env!("OUT_DIR"), "/multisig.wasm"));
 
 #[iroha_trigger::main]
-fn main(_id: TriggerId, _owner: AccountId, event: EventBox) {
-    let args: MultisigRegisterArgs = match event {
-        EventBox::ExecuteTrigger(event) => event
-            .args()
-            .dbg_expect("trigger expect args")
-            .try_into_any()
-            .dbg_expect("failed to parse args"),
-        _ => dbg_panic("Only work as by call trigger"),
+fn main(host: Iroha, context: Context) {
+    let EventBox::ExecuteTrigger(event) = context.event else {
+        dbg_panic("Only work as by call trigger");
     };
 
+    let args: MultisigRegisterArgs = event
+        .args()
+        .try_into_any()
+        .dbg_expect("failed to parse args");
+
     let account_id = args.account.id().clone();
 
-    Register::account(args.account)
-        .execute()
+    host.submit(&Register::account(args.account))
```

Comment:
the unfortunate difference from `iroha` client is that we take instruction by reference instead of value